### ISSUE TITLE
Automatisation de la création d'issues sur github

### ISSUE THREAD

**Issue by Goutte** [jlm2017/jlm-video-subtitles#207]
Bonjour !

Comme [la première sauvegarde automatisée](https://github.com/jlm2017/jlm-video-subtitles/commit/5884903fa38f737314910a8bb8498de2484bd139) :v: a bien fonctionné, je commence à regarder ce qui est dans mes cordes pour la suite...

Automatiser la création d'issues sur github, c'est faisable, techniquement. Ainsi que la [création des "cartes" dans un projet](https://developer.github.com/v3/projects/cards), et l'assignation de labels.

# Pourquoi

J'entends déjà les râleurs râler : _ah, ces saletés d'algorithmes qui nous volent notre travail, grmbl, et y parlent même pas le français..._

Plus sérieusement, quand on regarde la liste des issues, il y a des erreurs (bon, on n'en voit pas trop, c'est parce que je fais du nettoyage), des casses différentes, des conventions différentes, des _peluches_ comme on dit dans le métier. Si on automatise, tout sera niquel et ça sera plus facile de s'y retrouver via la recherche, sans compter que ça prépare le terrain pour d'autres hypothétiques algorithmes qui liraient les issues...

Bref, c'est toujours mieux quand c'est bien rangé, dans le domaine de l'informatique.
Et puis, franchement, c'est pas ça qui est le plus intéressant à faire, si ?

# Comment

- Pour quelles langues créer automatiquement les issues ? _Français, anglais ?_
- Pour quelles vidéos créer automatiquement les issues ? _Je dirais seulement pour les nouvelles ?_
- A quelle fréquence doit-on vérifier la présence de nouvelles vidéos, et créer les issues en conséquence ? _Il y a des quotas à respecter, donc pas trop souvent non plus ; toutes les heures ?_
- A-t-on vraiment besoin du `[subtitles]` dans le titre de l'issue ?

# Quand

Quand on sera d'accord sur quoi faire ; le plus tôt est le mieux, non ?

# Qui

Bibi ; sauf si quelqu'un d'autre veut le faire, évidemment.
Techniquement, c'est [RobotInsoumis](https://github.com/RobotInsoumis) qui créera les issues.

# Où

Ici, sur github. Le forum phpbb de M. Paroche, c'est compliqué, il est trop à la ramasse technologiquement _(le forum, s'entend)_.

**Comment by LucileDT** [comment]
Hello @Goutte ! Merci pour tes propositions, c'est top :+1: 

Pour ma part, je dirais : 
- vérifier toutes les heures
- automatiser pour les langues FR, EN, et DE (puisqu'on commence à avoir les germanophones qui bossent dessus), quitte à enlever DE si on se rend compte que finalement, c'est catastrophique
- une fois par semaine, rajouter une "vieille" vidéo dans les issues, en partant des plus récentes (comme ça on finira par bien tout avoir, sans que ce soit trop stressant)
- on peut enlever [subtitles] si vous le souhaitez :)

### Maintenant, des questions : 
- est-ce qu'au moment de la création de l'issue Github, il y a moyen de créer dans la vidéo un sous-titre ?
- dans le sous-titre, vous voudriez mettre quoi ? Je verrais bien une petite phrase du style 
> "Le groupe d'insoumis transcripteurs se charge de cette vidéo, vous pouvez constater l'avancement du travail à : <url de l'issue>" ?

- est-ce que tu crois qu'il sera possible à un moment de mettre à jour les labels automatiquement sur les issues ? Ce serait super, mais j'ai peur que ce soit très compliqué.

**Comment by Goutte** [comment]
Merci @LucileDT pour cette réponse rapide !

> une fois par semaine, rajouter une "vieille" vidéo dans les issues, en partant des plus récentes (comme ça on finira par bien tout avoir, sans que ce soit trop stressant)

C'est une bonne idée ! Je note !

> est-ce qu'au moment de la création de l'issue Github, il y a moyen de créer dans la vidéo un sous-titre ?

Absolument ! Avec la petite phrase, et tout ; mais il me faut les permissions sur la chaîne YouTube :see_no_evil: ! Donc, en pratique, non.

> est-ce que tu crois qu'il sera possible à un moment de mettre à jour les labels automatiquement sur les issues ? Ce serait super, mais j'ai peur que ce soit très compliqué.

Je ne vois pas vraiment comment ? En fait, tant que YouTube ne permet pas l'édition collaborative en temps réel... Je ne sais pas trop ce qu'il se passe quand on est plusieurs à modifier un sous-titre en même temps, mais d'après vos dires ce n'est pas la joie pour l'instant. Il faudra que je fasse des tests.

Edit: finalement, [c'est fait](https://github.com/jlm2017/jlm-video-subtitles/issues/229).

**Comment by LucileDT** [comment]
@Goutte concernant la petite phrase, j'ai une précision : 
- s'il n'y a pas de sous-titres, mettre la petite phrase,
- s'il y a des sous-titres, préciser dans l'issue Github que quelqu'un a déjà commencé à travailler dessus à l'extérieur (je peux créer un label pour ça).

Est-ce que tu as pu faire des tests avec ma chaîne ? Désolée, j'ai pas check les mails depuis un moment.

**Comment by Goutte** [comment]
@LucileDT  Ça me semble raisonnable, comme comportement.

J'ai fait les tests que j'avais besoin de faire, oui, et merci ! :heart:

**Comment by LucileDT** [comment]
Donc tu confirmes avoir besoin d'être dans les administrateurs de la chaîne ?

**Comment by Goutte** [comment]
Oui, ça ou avoir directement les identifiants de la chaîne. :key:

**Comment by Goutte** [comment]
L'issue #215 est la première issue créée automatiquement. C'est un test, il n'y en aura pas d'autres tant qu'il n'y aura pas de feu vert.

---

J'ai essayé de mettre en place un tableau plutôt que des titres, c'est plus correct d'un point de vue sémantique. Et puis c'est fastidieux à faire à la main, mais comme là c'est un script qui s'en charge...

Bref, vous me dites ; je préfère un tableau, mais je ne veux pas bousculer vos habitudes trop fort.

---

Je ne peux pas ajouter automatiquement une _Card_ au _Project_ ; 

```
403 Must have admin rights to Repository.
```

Surprise, suprise ! Encore des problèmes de permissions ! :wink:

**Comment by Bentham** [comment]
C'est bien propre cette issue #215 ! Bravo Goutte, j'admire tes super-pouvoirs de geek !

**Comment by Goutte** [comment]
Je ne sais pas pourquoi le lien vers l'éditeur n'est pas fonctionnel, j'ai fait comme propose @LucileDT dans son template ?

---

Edit: Ah, c'est peut-être à cause du paramètre `forceedit=timedtext`...

**Comment by Goutte** [comment]
@LucileDT Il est fort probable que m'ajouter [au groupe](https://github.com/jlm2017) suffise.

**Comment by Goutte** [comment]
@LucileDT On ne veut pas m'ajouter en tant qu'administrateur au groupe susmentionné pour des raisons de _sécurité_. M'ajouter comme administrateur à ce dépôt est faisable, à toi de voir si tu me fais confiance ou pas. :lock: 

---

C'est ce que doit ressentir JLM parfois ; avoir tout qui est prêt sur le fond, et ne pas pouvoir le faire à cause du manque de confiance des gens.

**Comment by RobotInsoumis** [comment]
Bonjour ! Je suis un :robot: insoumis !

Je serai le bot qui va effectuer les sauvegardes et créer les issues, dès que @LucileDT m'aura ajouté aux administrateurs de ce dépôt !

_(ça sera bien plus propre qu'avoir @Goutte de partout)_

**Comment by LucileDT** [comment]
@Goutte @RobotInsoumis vous faites désormais partie des administrateurs du dépôt :D

**Comment by Goutte** [comment]
It works ! :tada: :tada: :tada: :balloon: :balloon: :balloon: :tada: :tada: :tada: 

Je lance ?

**Comment by LucileDT** [comment]
Je t'en prie :)

**Comment by Goutte** [comment]
Bon, c'est parti !

Il reste :
1. Ajouter les vieilles vidéos au rythme de 1 par semaine _(c'est assez compliqué, en fait)_
2. Ajouter l'allemand _(je l'ai mis, puis enlevé ; germanophones, manifestez-vous!)_
3. Uploader un brouillon sur youtube avec un petit message _(need permissions)_

---

J'ai hâte que JLM fasse une nouvelle vidéo, pour confirmer que tout fonctionne ! :trepigne:

**Comment by LucileDT** [comment]
@Goutte Pourrait-on revoir ensemble la mise en page des issues ? Je les trouve pour ma part difficilement lisibles, mais c'est peut-être parce que c'est complètement différent de ce qui existait avant.

**Comment by Goutte** [comment]
:sob: @LucileDT Je suis à votre disposition aujourd'hui sur

1. https://meet.jit.si/insoumis%E2%8B%85es
2. Discord (Goutte dans le Saloon des Insoumis)

---

J'aime bien les tableaux, moi. :sob:

**Comment by Goutte** [comment]
OK les gens, la nouvelle mise en forme c'est [par exemple ici](https://github.com/jlm2017/jlm-video-subtitles/issues/224).

**Comment by LucileDT** [comment]
@Goutte peux-tu éditer l'issue #1 plutôt que d'en recréer une ? Merciiii
